### PR TITLE
Add tests for SerialNumber arithmetic (RFC 1982)

### PR DESCRIPTION
**Pull request by ccerdadiaz**
serial_number.rs had no test coverage (0%). This PR adds unit tests for SerialNumber arithmetic as defined in RFC 1982.

 Test covering:
- Equality
- Addition with wrapping (u32 overflow)
- Less than: normal case and wrap-around case
- Greater than: normal case and wrap-around case
- Undefined case (None) when distance equals 2^31

### CODE REVIEW

**Review by divergentdave — APPROVED** [review]
FWIW, I ran `just coverage-lcov` and it seems this file has 72% test coverage, due to it getting exercised elsewhere in DNSSEC tests. Note that these are only enabled with certain non-default Cargo features.

These tests look reasonable, it would be good to have tests specifically covering these edge cases.

File: crates/proto/src/rr/serial_number.rs [hunk 1]
```diff
@@ -65,3 +65,57 @@ impl PartialOrd for SerialNumber {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_equal() {
```
Comment (djc) [inline crates/proto/src/rr/serial_number.rs]: Nit: suggest dropping the `test_` prefixes from these.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
(And please make sure to squash all commits into one.)

**Review by djc — APPROVED** [review]
